src/components/ToolMenu/LayerTree/index.tsx

```diff
@@ -72,11 +74,15 @@ export const LayerTree: React.FC<LayerTreeProps> = ({
     t
   } = useTranslation();
 
-  const [visibleLegendsIds, setVisibleLegendsIds] = useState<string[]>([]);
-  const [layerTileLoadCounter, setLayerTileLoadCounter] = useState<LayerTileLoadCounter>({});
+  const showLegendsState = useAppSelector(state => state.layerTree.showLegends);
+  // check if showLegends is defined
+  const showLegends: boolean = showLegendsState ? showLegendsState : false;
```

Comment:
Is this really needed? Can't we use `showLegendsState` directly?

Reply:
Only use of showLegendsState in the new commit